pkg/java/pom/pom.go

```diff
@@ -25,7 +25,12 @@ func (p *pom) inherit(result analysisResult) {
 
 	p.content.GroupId = art.GroupID
 	p.content.ArtifactId = art.ArtifactID
```

Comment:
Can't artifactId be a property?

Reply:
I have not seen information or examples of this.
Property in `Version` is needed to change version for all nested pom files.
For `ArtifactID` this doesn't make much sense, because you don't need to change `ArtifactID` from time to time. You create `GroupID` and `ArtifactID` once and use them.

Reply:
https://maven.apache.org/maven-ci-friendly.html
`Starting with Maven 3.5.0-beta-1 you can use the ${revision}, ${sha1} and/or ${changelist} as placeholders for the version in your pom file.`

Reply:
hm.. it is strange. This works, but maven returns a warning about this case:
pom.xml:
```
...
    <artifactId>${revision}-test-revision</artifactId>    
...
```
maven:
```
➜  ali-dbhub-server-domain-support mvn dependency:tree | grep test-revision
[WARNING] Some problems were encountered while building the effective model for com.alibaba.dbhub:1.0.0-SNAPSHOT-test-revision:jar:1.0.0-SNAPSHOT
[WARNING] 'artifactId' contains an expression but should be a constant. @ com.alibaba.dbhub:${revision}-test-revision:${revision}, /home/dmitriy/work/temp/4670/Chat2DB-bd1edda4d485fab2e06632fb50716748353153f0/ali-dbhub-server/ali-dbhub-server-domain/ali-dbhub-server-domain-support/pom.xml, line 12, column 17
...
[INFO] com.alibaba.dbhub:1.0.0-SNAPSHOT-test-revision:jar:1.0.0-SNAPSHOT
```

Reply:
Thanks for the prompt testing! If maven warns it, we don't have to support it. I'll approve it.